### PR TITLE
Minor updates 

### PR DESCRIPTION
**Pull request by PavelGozhii**

### CODE REVIEW

File: Animal.java [hunk 1]
```diff
@@ -1,7 +1,9 @@
 package com.company;
 
-// Task 4 immutable class
+import jdk.nashorn.internal.ir.annotations.Immutable;
```
Comment (Siril-O) [inline Animal.java:3]: why do you  need this import ?

File: Animal.java [hunk 2]
```diff
@@ -17,11 +19,19 @@ public String getName() {
...
         return age;
     }
 
-    public double getHeight(){
+    public double getHeight() {
         return height;
     }
+
+
+    //As I understand it, we need to change the class field without changing it,
```
Comment (Siril-O) [inline Animal.java:31]: yes, but remove your comment

File: Builder.java [hunk 3]
```diff
@@ -2,17 +2,30 @@
 
 public class Builder {
 
-    public Shape builTriangle(int[] lengthOfSides) {
+
+    public Shape buildTriangle(int[] lengthOfSides) throws InvalidInputException {
+        //Checking valid input
```
Comment (Siril-O) [inline Builder.java:7]: remove comment

File: Builder.java [hunk 4]
```diff
@@ -2,17 +2,30 @@
...
-        }else return null;
+            if (lengthOfSides[0] > 0 && lengthOfSides[1] > 0 && lengthOfSides[2] > 0) {
+                if ((lengthOfSides[0] + lengthOfSides[1]) < lengthOfSides[2] &&
+                        lengthOfSides[1] + lengthOfSides[2] < lengthOfSides[0] &&
+                        lengthOfSides[2] + lengthOfSides[0] < lengthOfSides[1]) {
+                    return new Triangle(lengthOfSides);
+                }
+            }
+        }
+        throw new InvalidInputException();
```
Comment (Siril-O) [inline Builder.java:17]: add some error message to exception

File: Drink.java [hunk 5]
```diff
@@ -1,48 +1,62 @@
...
-        private String name;
-        private String description;
-        private double price;
+    @Override
+    public String toString(){
+        return name + "\n" + description + "\n" + price;
+    }
+}
+
+class DrinkBuilder{
```
Comment (Siril-O) [inline Drink.java:34]: private static builder was ok

Comment (Siril-O) [reply Drink.java:34]: move it back and create static method  in Drink to get instance of this builder

File: Drink.java [hunk 6]
```diff
@@ -1,48 +1,62 @@
...
+    private String name = "";
```
Comment (Siril-O) [inline Drink.java:35]: Why empty string ?

File: Sort.java [hunk 7]
```diff
@@ -22,36 +22,38 @@
...
         int left = 0;
         int right = arr.length - 1;
+        int temp;
         do {
             for (int i = left; i < right; i++) {
                 if (arr[i] > arr[i + 1]) {
-                    arr[i] = arr[i] ^ arr[i + 1];
-                    arr[i + 1] = arr[i] ^ arr[i + 1];
-                    arr[i] = arr[i] ^ arr[i + 1];
+                    temp = arr[i];
```
Comment (Siril-O) [inline Sort.java:29]: good, now put it to separate method not to have duplicates

**Review by Siril-O — CHANGES_REQUESTED** [review]

File: Animal.java (hunk 1 above)
Comment (PavelGozhii) [reply Animal.java:3]: t

> why do you need this import ?

to put an immutable annotation

File: Drink.java (hunk 6 above)
Comment (PavelGozhii) [reply Drink.java:35]: > Why empty string ?

default value

**Pull request merged**